devito/petsc/solve.py

```diff
@@ -66,8 +75,9 @@ def separate_eqn(eqn, target):
     zeroed_eqn = Eq(eqn.lhs - eqn.rhs, 0)
     tmp = eval_time_derivatives(zeroed_eqn.lhs)
     b = remove_target(tmp, target)
-    F_target = simplify(tmp - b)
-
+    # Is this ok? Is there another way of using simplify
```

Comment:
You can use `devito.finite_differences.differentiable.diffify` to turn `sp.Add` etc into `dv.Add`

Reply:
why do you need simplify, what's wrong with just `tmp - b` ?

Reply:
In some cases this does not work and it evaluates to `-term + term` , but `simplify` evaluates it to `0`